### ISSUE TITLE
Layers created with tf.layers not listed with tf.contrib.framework.get_model_variables

### ISSUE THREAD

**Issue by Tauranis** [tensorflow/tensorflow#16594]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.4.0
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: Not Applicable
- **GCC/Compiler version (if compiling from source)**: Not Applicable
- **CUDA/cuDNN version**: 8/6
- **GPU model and memory**: GeForce 940MX

### Describe the problem
Currently I'm using [slim models](https://github.com/tensorflow/models/tree/master/research/slim#pre-trained-models) to build my networks, but for extra layers I'm using the [tf.layers](https://www.tensorflow.org/api_docs/python/tf/layers) API. When I try to retrieve the list of variables with the function [`tf.contrib.framework.get_model_variables`](https://www.tensorflow.org/versions/master/api_docs/python/tf/contrib/framework/get_model_variables)
only the layers created inside slim models are retrieved.

### Minimal code to reproduce the issue

```python
import tensorflow as tf
x = tf.random_normal([2,5],name='x')
y = tf.contrib.layers.fully_connected(x,10,scope='Contrib_Fully_Connected') # Retrieved
z = tf.layers.dense(x,10, name='Layers_Fully_Connected') # Not Retrieved
print(tf.contrib.framework.get_model_variables())
```

Output:

```bash
[<tf.Variable 'Contrib_Fully_Connected/weights:0' shape=(5, 10) dtype=float32_ref>, <tf.Variable 'Contrib_Fully_Connected/biases:0' shape=(10,) dtype=float32_ref>]
```

### Workaround:

Use [`tf.get_collection`](https://www.tensorflow.org/api_docs/python/tf/get_collection)

```python
import tensorflow as tf
x = tf.random_normal([2,5],name='x')
y = tf.contrib.layers.fully_connected(x,10,scope='Contrib_Fully_Connected') # Retrieved
z = tf.layers.dense(x,10, name='Layers_Fully_Connected') # Not Retrieved
print(tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES))
```
the output is:
```bash
[<tf.Variable 'Contrib_Fully_Connected/weights:0' shape=(5, 10) dtype=float32_ref>,
 <tf.Variable 'Contrib_Fully_Connected/biases:0' shape=(10,) dtype=float32_ref>,
 <tf.Variable 'Layers_Fully_Connected/kernel:0' shape=(5, 10) dtype=float32_ref>,
 <tf.Variable 'Layers_Fully_Connected/bias:0' shape=(10,) dtype=float32_ref>]
```

Is it worth to bring this issue as a bug and improve the funcionality of `tf.contrib.framework.get_model_variables` or will this contrib function be discontinued?

**Comment by drpngx** [comment]
I think it's worth sending a fix if you know what the problem is.

**Comment by imsheridan** [comment]
I am not sure whether this is by design since in [get_model_variables](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/framework/python/ops/variables.py#L338) it set the scope as ops.GraphKeys.MODEL_VARIABLES instead of tf.GraphKeys.GLOBAL_VARIABLES as @Tauranis did for tf.get_collection above. 
@drpngx, could you please help confirm this?

**Comment by drpngx** [comment]
I see.

@martinwicke do you have more context on the collection names and which one we're supposed to be using?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by martinwicke** [comment]
@allenlavoie is working on a better way of getting to variables. Even now, I would generally prefer the .variables property of layer objects.

However, this is a bug, and we could add code to tf.layers.Layer to fix this. Marking contributions welcome.
